torchinductor/codegen/autotuner.py

```diff
@@ -0,0 +1,166 @@
+import builtins
+
+import torch
+import triton
+
+import torchinductor
+import torchinductor.triton_ops
+
+aten = torch.ops.aten
+triton_ops = torchinductor.triton_ops
+
+# Instatiatie Autotuner class object -- init the cache, and sort of global
+
+# Autotuning an op
+#   1) Register that Node in the codegen of ExternKernelOP
+#   2) Autotuner
+#   1) register that IRNode "ExternKerneNode" in Autotuner - autotune()
+#
+# def autotune():
+#     """
+#     Decorator for auto-tuning a function.
+
+#     .. highlight:: python
+#     .. code-block:: python
+
+#         @triton.autotune()
+#         wrap_conv(x, w, bias, stride, ...)
+
+#     :note: For the wrap function, the candidate kernels will run multiple time.
+#     """
+#     def inner(fn):
+#         autotuner = Autotuner(fn)
+#         return autotuner
+
+#     return inner
+
+
+class Autotuner:
+    def __init__(self):
+
+        self.cache = dict()
+
+    def _bench(self, *args, kernel, **kwargs):
+        def kernel_call():
+            kernel(*args, **kwargs)
+
+        return triton.testing.do_bench(kernel_call, warmup=10, rep=50)
+
+    # def __call__(self, *args, **kwargs):
+    #     # collect input shape/stride/device, layer params
+    #     self.fn.set_args(*args, **kwargs)
+    #     # get candidate kernels for fn
+    #     self.kernels = self.fn.candidate_kernels()
+    #     # filter kernels that args/kwargs does not meet requirements
+    #     self.fn.filter_kernels(self.kernels)
+    #     # if only one choice, return that kernel
+    #     if len(self.kernels) == 1:
+    #         kernel = self.kernels[0]
+    #         return kernel(*args, **kwargs)
+    #     # use input shape, stride, layer parameters as key to cache the best kernel to use
+    #     key = self.fn.gen_key()
+    #     if key not in self.cache:
+    #         bench_start = time.time()
+    #         timings = {
+    #             kernel: self._bench(*args, kernel=kernel, **kwargs) for kernel in self.kernels
+    #         }
+    #         bench_end = time.time()
+    #         self.bench_time = bench_end - bench_start
+    #         self.cache[key] = builtins.min(timings, key=timings.get)
+    #         self.kernels_timings = timings
+    #         kernel = self.cache[key]
+    #     self.best_kernel = kernel
+    #     return self.best_kernel(*args, **kwargs)
+
+
+autotune = Autotuner()
+
+
+def tuned_conv(*args, **kwargs):
+
+    # gather info
+    x = args[0]
```

Comment:
Can we use normal arg proessing here rather than args/kwargs.  If the user actually uses kwargs this work work right.